src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -355,10 +356,41 @@ static QString AnalyzeURI( QString const &uri,
 
       else if ( field == QLatin1String( "geometrytype" ) )
       {
-        ogrGeometryTypeFilter = ogrWkbGeometryTypeFromName( value );
+        ogrGeometryTypeFilter = QgsOgrProviderUtils::wkbGeometryTypeFromName( value );
+      }
+      else if ( field == QLatin1String( "geometryname" ) )
+      {
+        geometryName = value;
+      }
+      else if ( field == QLatin1String( "geometryid" ) )
+      {
+        geometryIndex = value.toInt( &ok );
+      }
+      else if ( field == QLatin1String( "featurescounted" ) )
+      {
+        if ( value != QLatin1String( "Unknown" ) )
+        {
+          featuresCounted = value.toInt( &ok );
+        }
+      }
+      else if ( field == QLatin1String( "ogrtype" ) )
+      {
+        ogrType = value.toInt( &ok );
       }
     }
-
+    if ( geometryName == QString::null )
+    {
+      if ( ( layerName.contains( "(" ) ) && ( layerName.endsWith( QString( ")" ) ) ) )
```

Comment:
> table_name(field_name) is only valid as a OGR layer name for the SQLite driver and this is a legacy behaviour of the GDAL 1 era. That will *not* work for other formats that can have multiple geometry fields, like GML.
 
The goal here was that **if** it was used (by a user in python)
* this would be corrected to avoid an invalid layer message

Sample:
`vl_polygons = QgsVectorLayer(u'Rauletshof_1909.gml|layername=SELECT(geometry_polygon)', u'gml', u'ogr')`

but should be:

`vl_polygons = QgsVectorLayer(u'Rauletshof_1909.gml|layername=SELECT|geometryname=geometry_polygon', u'gml', u'ogr')`

The goal is to make it **user friendly**, applying an 'autocorrection' where practical
* avoiding failures